services/consensusalgo/benchmarkconsensus/test/non_leader_test.go

```diff
@@ -2,39 +2,135 @@ package test
 
 import (
 	"context"
-	"github.com/orbs-network/go-mock"
 	"github.com/orbs-network/orbs-network-go/test"
 	"github.com/orbs-network/orbs-network-go/test/builders"
-	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
-	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"testing"
 )
 
 func TestNonLeaderDoesNotCreateBlocks(t *testing.T) {
 	test.WithContext(func(ctx context.Context) {
 		h := newHarness(false)
-		h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).Times(0)
+		h.expectNoBlockCreation()
 		h.createService(ctx)
-		err := test.ConsistentlyVerify(h.consensusContext)
-		if err != nil {
-			t.Fatal("Did create block with ConsensusContext:", err)
-		}
+		h.verifyNoBlockCreation(t)
 	})
 }
 
-func TestNonLeaderIgnoreFutureBlockHeight(t *testing.T) {
+func TestNonLeaderCommitsAndRepliesToValidBlockHeights(t *testing.T) {
```

Comment:
I don't like this test:
* I don't understand why you need to assert on 3 blocks
* It takes time to understand that you're asserting on a long-going story
* Maybe this needs to be a unit test?

Reply:
Originally it was two tests:

1. Test that two commits in succession changes the reply

2. Test that an out of order commit (old one) doesn't change the reply

Because test 2 was exactly test 1 with an added block at the end, I removed test 1 to avoid duplication.

I don't want to move this to a unit test because it's part of the component spec

Do you want me to split it back to 2 tests maybe?